src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,82 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import { FC, useEffect, useState } from 'react';
+import { Todo } from '../../types/Todo';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+interface Props {
+  todos: Todo[];
+  setSortedTodos: (todo: Todo[]) => void;
+}
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+export const TodoFilter: FC<Props> = ({
+  setSortedTodos,
+  todos,
+}) => {
+  const [search, setSearch] = useState('');
+  const [sortBy, setSortBy] = useState('all');
+
+  const filter = () => {
+    switch (sortBy) {
+      case 'active':
+        setSortedTodos([...todos].filter(todo => !todo.completed
+          && todo.title.includes(search)));
```

Comment:
don't repeat code. You can create separate method to check includes search in title